src/numerics/petsc_matrix.C

```diff
@@ -725,7 +725,7 @@ void PetscMatrix<T>::print_personal(std::ostream & os) const
   // Print to screen if ostream is stdout
   if (os.rdbuf() == std::cout.rdbuf())
     {
-      ierr = MatView(_mat, PETSC_VIEWER_STDOUT_SELF);
+      ierr = MatView(_mat, NULL);
```

Comment:
I perused the docs for [MatView](https://www.mcs.anl.gov/petsc/petsc-current/docs/manualpages/Mat/MatView.html] briefly, but couldn't see what passing `NULL` for the second arg does?

Reply:
Thanks @jwpeterson ! I came across error in `MatView` about 'Arguments must have same communicator' while trying to print out a `PetscMatrix` for debugging purposes. @fdkong suggested me to make the two changes in `print_personal`, which resolved the issue.

@fdkong Would you mind explaining a little more on what the NULL does here?

Reply:
`NULL` is the right option to use because we want Mat print out on the screen for both parallel and serial cases.  Here are codes:

```
975: PetscErrorCode MatView(Mat mat,PetscViewer viewer)
976: {
977:   PetscErrorCode    ierr;
978:   PetscInt          rows,cols,rbs,cbs;
979:   PetscBool         isascii,isstring,issaws;
980:   PetscViewerFormat format;
981:   PetscMPIInt       size;

986:   if (!viewer) {PetscViewerASCIIGetStdout(PetscObjectComm((PetscObject)mat),&viewer);}
```

Reply:
OK, makes sense to me, I was not aware of this `if (!viewer)` logic.